bootstrap/eks/internal/userdata/utils.go

```diff
@@ -32,3 +38,28 @@ func templateYAMLIndent(i int, input string) string {
 	ident := "\n" + strings.Repeat(" ", i)
 	return strings.Repeat(" ", i) + strings.Join(split, ident)
 }
+
+func templateToYAML(r *runtime.RawExtension) (string, error) {
+	if r == nil {
+		return "", nil
+	}
+	if r.Object != nil {
+		b, err := yaml.Marshal(r.Object)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to convert to yaml")
+		}
+		return string(b), nil
+	}
+	if len(r.Raw) > 0 {
```

Comment:
Please `if len(r.Raw) == 0 { return "", nil` early instead of adding one level of indentation